be/src/exprs/agg/nullable_aggregate.h

```diff
@@ -228,21 +235,32 @@ class NullableAggregateFunctionUnary final : public NullableAggregateFunctionBas
                 // TODO(kks): when our memory allocate could align 32-byte, we could use _mm256_load_si256
                 __m256i f = _mm256_loadu_si256(reinterpret_cast<const __m256i*>(f_data + offset));
                 int mask = _mm256_movemask_epi8(_mm256_cmpgt_epi8(f, all0));
-                // all not null
                 if (mask == 0) {
+                    // all not null
                     for (size_t i = offset; i < offset + batch_nums; i++) {
                         this->data(states[i] + state_offset).is_null = false;
                         this->nested_function->update(ctx, &data_column,
                                                       this->data(states[i] + state_offset).mutable_nest_state(), i);
                     }
-                    // all null
                 } else if (mask == 0xffffffff) {
+                    // all null
+                    if constexpr (!IgnoreNull) {
+                        for (size_t i = offset; i < offset + batch_nums; i++) {
+                            this->data(states[i] + state_offset).is_null = false;
+                            this->nested_function->update_null(
+                                    ctx, this->data(states[i] + state_offset).mutable_nest_state());
+                        }
+                    }
                 } else {
                     for (size_t i = offset; i < offset + batch_nums; i++) {
                         if (f_data[i] == 0) {
                             this->data(states[i] + state_offset).is_null = false;
                             this->nested_function->update(ctx, &data_column,
                                                           this->data(states[i] + state_offset).mutable_nest_state(), i);
+                        } else if constexpr (!IgnoreNull) {
```

Comment:
Why when the input is null, make the state is_null to false?

Reply:
```
mysql> select * from m3;
+------+------+
| a    | b    |
+------+------+
|    1 | NULL |
|    2 | NULL |
+------+------+
2 rows in set (0.93 sec)

mysql> select array_agg(b) from m3 group by a;
+----------------+
| array_agg(`b`) |
+----------------+
| [null]         |
| [null]         |
+----------------+
2 rows in set (0.07 sec)

if not set is_null=false, it will be

mysql> select array_agg(b) from m3 group by a;
+----------------+
| array_agg(`b`) |
+----------------+
| null         |
| null         |
+----------------+
2 rows in set (0.07 sec)
```

Reply:
Set  is_null=false, which only right for `array_agg`. If one aggregate function doesn't ignore null and only output null when all input is null, how do you handle it?

Reply:
Not just designed for this function...

is_null = false: means not discard the values ...

```
If one aggregate function doesn't ignore null and only output null when all input is null, how do you handle it?

orginal design can handle it, e.g. sum
```

Reply:
is_null = false means there is a value, is_null = true means there is no value, or the  value is all discarded